fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/MetadataManager.java

```diff
@@ -169,6 +170,28 @@ public Map<String, Long> listPartitions(TablePath tablePath)
                 "Fail to list partitions for table: " + tablePath);
     }
 
+    /**
+     * List the partitions of the given table and partitionSpec.
+     *
+     * <p>Return a map from partition name to partition id.
+     */
+    public Map<String, Long> listPartitions(
```

Comment:
This is inner class, please combine this method with `public Map<String, Long> listPartitions(TablePath tablePath)
            throws TableNotExistException, TableNotPartitionedException` to avoid duplicated data.